unstructured/ingest/connector/vectara.py

```diff
@@ -0,0 +1,246 @@
+import datetime
+import json
+import traceback
+import typing as t
+import uuid
+from dataclasses import dataclass
+
+import requests
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.interfaces import (
+    AccessConfig,
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.staging.base import flatten_dict
+
+
+@dataclass
+class VectaraAccessConfig(AccessConfig):
+    oauth_client_id: str = enhanced_field(sensitive=True)
+    oauth_secret: str = enhanced_field(sensitive=True)
+
+
+@dataclass
+class SimpleVectaraConfig(BaseConnectorConfig):
+    access_config: VectaraAccessConfig
+    customer_id: t.AnyStr = None
```

Comment:
If we're setting this to a default of `None`, the typing should be `t.Optional[]`. But should this be optional? Or are these required? Also why are we supporting `bytes`? Which is part of the `AnyStr` constraint.

Reply:
customer_id should not be optional. corpus_name and corpus_id should be. I thought AnyStr was interesting. I saw that it included bytes. But I think we just need `str`